Source/JavaScriptCore/wasm/WasmLLIntGenerator.cpp

```diff
@@ -2000,17 +2000,41 @@ auto LLIntGenerator::addStructNew(uint32_t index, Vector<ExpressionType>& args,
 {
     result = push();
 
+    // We have to shift all the arguments up the stack by one slot, because we just
```

Comment:
Can we add a new dummy argument to avoid this?

Reply:
I don't think that adding an extra argument helps. Suppose we enter `addStructNew()` with 3 arguments and the stack depth (before pushing the arguments) is 18. If these arguments are non-constants, they will already have been assigned to locations 19, 20, and 21. The dummy argument would either have to be 22 or 18, but there's no way to use 18 since that's deeper on the stack. If we use 22, the stack consistency checker will complain, or at least I couldn't find a way to do that without getting assertion failures.

I realized I could change it to just re-use the first argument for the return value (and special-case the 0-argument case). The previous version avoided overwriting any of the arguments, but I don't think that was necessary. The new version doesn't seem as clean, but all the tests pass.